pandas/core/internals/construction.py

```diff
@@ -271,7 +271,9 @@ def init_dict(data: Dict, index, columns, dtype: Optional[DtypeObj] = None):
                 nan_dtype = np.dtype(object)
             else:
                 nan_dtype = dtype
-            val = construct_1d_arraylike_from_scalar(np.nan, len(index), nan_dtype)
+
+            na_val = arrays[missing].iat[0]
```

Comment:
why not `na_value_for_dtype` ?

Reply:
`arrays` is always object-dtype, so that wont help.  the case of interest here is if we have `arrays = `pd.Series([np.timedleta64("NaT", "ns"), np.timedleta64("NaT", "ns")]), dtype=object)`.  We want na_val to be tdnat, not np.nan (which is what we use now)

Reply:
ok, then pls assert len(arrays[missing]) at the top and add a comment. also should this be arrays[missing].unique().item()?

Reply:
hmm need to give some thought to how to handle case where arrays[missing] isnt unique.  will gentle-ping when i figure this out

Reply:
updated to punt on the tougher case